app/controllers/assignments_controller.rb

```diff
@@ -92,7 +95,35 @@ def create
       @assignment.available = @assignment.due_date
     end
 
-    if @assignment.save
+    # Options are:
+    # None -- create a (unique, ignored) teamset
+    # New -- create a new teamset with initially no teams
+    # Use -- share an existing teamset with another assignment
+    # Copy -- duplicate an existing teamset
+    
+    @assignment.team_subs = (params["teamset"] != "none")
```

Comment:
This seems very dangerous to me: it implies that the model class can have temporary virtual state that is inconsistent. I've avoided changing the model's API so that you can't use the model to create a bogus teamset. Seems to me the logic really does belong in the controller, since the controller's methods are either called or not, and they'll either update the models or not, and it's ok for the controller to be complicated, if it's actually a delicate bit of control.